cmd/panfs-multipart.go

```diff
@@ -37,18 +37,19 @@ import (
 )
 
 const (
-	panbgAppendsDirName         = "bg-appends"
 	panbgAppendsCleanupInterval = 10 * time.Minute
 )
 
-// Returns EXPORT/.minio.sys/multipart/SHA256/UPLOADID
-func (fs *PANFSObjects) getUploadIDDir(bucket, object, uploadID string) string {
-	return pathJoin(fs.fsPath, minioMetaMultipartBucket, getSHA256Hash([]byte(pathJoin(bucket, object))), uploadID)
+// Returns EXPORT/bucket/.s3/multipart/SHA256/UPLOADID
+// TODO: refactoring. Add context to the arguments. Use getBucketPath from metadata. Remove bucketPath argument.
```

Comment:
About TODOs.
There are many usages of bucketPath near calls of this and the next function.
Won't it lead to additional needless calculations of the path?

Reply:
What exactly calculation you mean here? Calculation of `uploadIdDir` or bucket path itself?

Reply:
Bucket path itself.

If I understood TODO right, you want to put this code:
```
bucketPath, err := fs.getBucketPanFSPathFromMeta(ctx, bucket)
if err != nil {
    logger.LogIf(ctx, err, logger.Application)
    return
}
```
into two functions(getUploadIDDir, getMultipartSHADir).
But sometimes they are used together or code of parent function also uses bucket path.
So, in this case it looks like bucket path will be calculated several times in one call.

But maybe I got the idea in a wrong way, so could you explain?

Reply:
Got it - these comments are not valid for the moment - so I'll just remove them. 
You are right - it is not a good idea to put some kind of calculation inside these functions